### DISCUSSION TITLE
responding to request, keeping the connection alive and reading the stream

### DISCUSSION THREAD

**Discussion by RainerZufall187** [aio-libs/aiohttp#7670] [Q&A] [answered]
i need to respond to a request first and then read the client stream help

**Comment by Dreamsorcerer** [comment]
That's a bit vague, but depending on what you need, you might be able to use a 100-continue handler, or do something else with a StreamResponse, or use a websocket..

**Comment by RainerZufall187** [comment]
i tried returning streamresponse but its sending a 0 for some reason

**Reply by Dreamsorcerer** [reply]
You'd probably need to use .prepare() and .write(), for example.
https://docs.aiohttp.org/en/stable/web_reference.html#aiohttp.web.StreamResponse.prepare

**Comment by RainerZufall187** [comment]
> That's a bit vague, but depending on what you need, you might be able to use a 100-continue handler, or do something else with a StreamResponse, or use a websocket..

no i need to respond with 200

**Reply by Dreamsorcerer** [reply]
Sorry, you want to respond with 200, at which point the connection is closed, and then read from the client? That's obviously not going to work, the client will have closed the connection after receiving the response.

Again, your description is vague, try describing what you are trying to achieve in more detail.

**Reply by RainerZufall187** [reply]
i want to make an icecast server and it works like that

**Reply by RainerZufall187** [reply]
```py
response = web.StreamResponse()
response._keep_alive = True
return response
```

**Reply by Dreamsorcerer** [reply]
Ah, see https://github.com/aio-libs/aiohttp/discussions/7086

If you need to support the outdated implementation mentioned there, then aiohttp is not the tool for you. This is an HTTP library, while the old icecast protocols are not HTTP...

**Reply by RainerZufall187** [reply]
ok

**Reply by Dreamsorcerer** [reply]
According to that information, the newer icecast protocol uses 100-continue, and would probably work with aiohttp. But, older versions are unlikely to work and you'd be better writing integration at a lower level.

**Reply by RainerZufall187** [reply]
so there is no way to keep the connection open after i do request.transport.write(b"HTTP/1.0 200 OK\r\n\r\n") in the handler and return nothing?

**Reply by Dreamsorcerer** [reply, marked as answer]
Maybe, I'm not sure. But, it's getting complicated and hacky, so you'd be better off just using low-level asyncio objects, like the previous user did, rather than trying to contort aiohttp to do something it's not designed for.
